docs/pages/docs/features/filtering.mdx

```diff
@@ -0,0 +1,113 @@
+can also exclude targets. A target that matches any filter and is not explicitly excluded will
+be in the final entrypoint selection.
+
+Turborepo's filter syntax is based on pnpm's filter syntax, so it should feel familiar to pnpm users.
+
+### Filter by package
+
+Supports exact matches (`--filter=my-pkg`), and globs (`--filter=*my-pkg*`) are supported.
+Scoped packages (`@foo/bar`) can be abbreviated to just the package name (`bar`) as long as the package name
+is unique within the monorepo (e.g. `@foo/bar` exists, but `@baz/bar` does not).
+
+```sh
+turbo run build --filter=my-pkg
+turbo run build --filter=@foo/bar
+turbo run build --filter=admin-*
+```
+
+### Include dependents of matched packages
+
+Prepend `...` to the filter. If `a` depends on `b`, `...b` will select `a` and `b`.
+Optionally including a `^` (`...^b`) will select all of `b`'s dependents, but not `b` itself. In this case, just `a` will be
+selected.
+
+```sh
+turbo run build --filter=...b
+turbo run build --filter=...^b
+```
+
+### Include dependencies of matched packages
+
+Append `...` to the end of the filter. If `a` depends on `b`, `a...` will select `a` and `b`.
+Optionally including a `^` (`a^...`) will select all of `a`'s dependencies, but not `a` itself. In this case, just `b` will be selected.
+
+```sh
+turbo run build --filter=a...
+turbo run build --filter=a^...
+```
+
+### Filter by directory or directory tree
+
+Supports exact matches (`--filter=./apps/docs`) and globs (`--filter=./apps/*`).
+When combined with other components, must be enclosed in `{}` (`--filter=...{./libs/*}`).
+
+```sh
+turbo run build --filter=./apps/*
+turbo run build --filter=...{./libs/*}
+```
+
+### Filter by changed packages
+
+Use the set of files changed since a specified commit to calculate packages. Enclose references in
+`[]`. For example, `--filter=[HEAD^1]` will select all packages that have changed in the most recent
+commit.
+
+You can use [`--ignore`](/docs/reference/command-line-reference#--ignore) to specify changed files to be ignored in the calculation of which packages have changed.
+
+You can additionally prepend the commit reference with `...` to match the dependencies of other components
+against the changed packages. For instance, to select `foo` if any of `foo`'s dependencies have changed in the last commit,
+you can pass `--filter=foo...[HEAD^1]`. Note that this feature is different from `pnpm`'s syntax.
+
+```sh
+turbo run build --filter=[HEAD^1]
+turbo run build --filter=...[origin/my-feature]
+turbo run build --filter=@foo/bar...[HEAD^1]
+```
+
+### Excluding packages
+
+Prepend `!` to the filter. Matched packages from the entire filter will be excluded from the set of targets.
+For example, match everything except `a`: `--filter=!a`. Note that you may need to escape `!` as appropriate for your shell (e.g. `\!`).
+
+```sh
+turbo run build --filter=!foo
+turbo run build --filter=./apps/* --filter=!./apps/admin
+```
+
+### Examples
```

Comment:
It seems like these examples might be better divided up in the sections that are the most relevant ☝️ just so the usage and examples are closer together.